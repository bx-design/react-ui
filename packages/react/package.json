{
  "name": "@bx-design/react-ui",
  "version": "1.0.0",
  "sideEffects": false,
  "license": "MIT",
  "repository": "git@github.com:bx-design/react-ui.git",
  "author": "BX Creative Studio",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "typings": "dist/types/index.d.ts",
  "exports": {
    ".": {
      "require": "./dist/cjs/index.js",
      "default": "./dist/esm/index.js"
    }
  },
  "files": [
    "dist"
  ],
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "lint": "yarn lint:src",
    "lint:src": "eslint src --ext .ts,.tsx --config ../../.eslintrc.json",
    "prebuild": "rimraf dist",
    "build": "concurrently yarn:build:*",
    "build:esm": "cross-env BABEL_ENV=esm babel src --root-mode upward --extensions .ts,.tsx -d dist/esm --source-maps",
    "build:cjs": "cross-env BABEL_ENV=cjs babel src --root-mode upward --extensions .ts,.tsx -d dist/cjs --source-maps",
    "build:types": "tsc -p ./tsconfig.json --declarationDir dist/types"
  },
  "dependencies": {
    "@bx-design/react-ui-alert": "^1.0.0",
    "@bx-design/react-ui-button": "^1.0.0",
    "@bx-design/react-ui-card": "^1.0.0",
    "@bx-design/react-ui-checkbox": "^1.0.0",
    "@bx-design/react-ui-dropdown": "^1.0.0",
    "@bx-design/react-ui-forms": "^1.0.0",
    "@bx-design/react-ui-modal": "^1.0.0",
    "@bx-design/react-ui-spinner": "^1.0.0",
    "@bx-design/react-ui-table": "^1.0.0",
    "@bx-design/react-ui-utils": "^1.0.0"
  },
  "devDependencies": {
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  },
  "peerDependencies": {
    "react": ">=17.0.2",
    "react-dom": ">=17.0.2"
  }
}
