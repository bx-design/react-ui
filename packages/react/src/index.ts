export * from '@bx-design/react-ui-alert'
export * from '@bx-design/react-ui-button'
export * from '@bx-design/react-ui-card'
export * from '@bx-design/react-ui-checkbox'
export * from '@bx-design/react-ui-dropdown'
export * from '@bx-design/react-ui-modal'
export * from '@bx-design/react-ui-spinner'
export * from '@bx-design/react-ui-table'
export * from '@bx-design/react-ui-forms'
export * from '@bx-design/react-ui-utils'
