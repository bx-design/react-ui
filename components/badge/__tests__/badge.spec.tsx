import React, { useState } from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import Badge, { VarianType } from '../badge'

describe('<Badge /> Component', () => {
  test('Render toMatchSnapshot', () => {
    const html = render(<Badge>Badge</Badge>)
    expect(html).toMatchSnapshot()
  })

  test('Should render component correctly', () => {
    render(<Badge>Badge</Badge>)
    const badge = screen.getByTestId('badge')
    expect(badge).toHaveTextContent('Badge')
  })

  test('Should render Component with props', () => {
    render(
      <Badge variant='primary' rounded>
        Badge
      </Badge>,
    )
    const badge = screen.getByTestId('badge')
    expect(badge).toHaveClass('badge')
    expect(badge).toHaveClass('badge-primary')
    expect(badge).toHaveClass('badge-rounded')
  })

  test('Should render Component with changing props', () => {
    function MockApp() {
      const [variant, setVariant] = useState<VarianType>('primary')
      const [rounded, setRounded] = useState(false)

      const handleClick = () => {
        setVariant('success')
      }

      return (
        <div>
          <Badge variant={variant} rounded={rounded}>
            Badge
          </Badge>
          <button onClick={handleClick}>Change Variant</button>
          <button onClick={() => setRounded(true)}>Change Rounded</button>
        </div>
      )
    }
    render(<MockApp />)
    const badge = screen.getByTestId('badge')
    const btnVariant = screen.getByText('Change Variant')
    const btnRounded = screen.getByText('Change Rounded')

    expect(badge).toHaveClass('badge')
    expect(badge).toHaveClass('badge-primary')
    userEvent.click(btnVariant)
    expect(badge).toHaveClass('badge-success')
    expect(badge).not.toHaveClass('badge-primary')
    userEvent.click(btnRounded)
    expect(badge).toHaveClass('badge-rounded')
  })
})
