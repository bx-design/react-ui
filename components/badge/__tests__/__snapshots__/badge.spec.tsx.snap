// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> Component Render toMatchSnapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="badge badge-primary"
        data-testid="badge"
      >
        <style>
          
            .badge {
              display: inline-block;
              padding: 0.35em 0.65em;
              font-weight: 700;
              line-height: 1;
              text-align: center;
              white-space: nowrap;
              vertical-align: baseline;
              border-radius: 0.25rem;
              color: var(--bx-white);
            }
            .badge-primary {
              background-color: var(--bx-orange);
            }
            .badge-secondary {
              background-color: var(--bx-grey-mistery);
            }
            .badge-success {
              background-color: var(--bx-green-future);
            }
            .badge-dark {
              background-color: var(--bx-black);
            }
            .badge-danger {
              background-color: var(--bx-red-medium);
            }
            .badge-info {
              background-color: var(--bx-lblue);
              color: var(--bx-black);
            }
            .badge-warning {
              background-color: var(--bx-yellow-alert);
              color: var(--bx-black);
            }
            .badge-rounded {
              border-radius: 50rem;
            }
          
        </style>
        Badge
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="badge badge-primary"
      data-testid="badge"
    >
      <style>
        
            .badge {
              display: inline-block;
              padding: 0.35em 0.65em;
              font-weight: 700;
              line-height: 1;
              text-align: center;
              white-space: nowrap;
              vertical-align: baseline;
              border-radius: 0.25rem;
              color: var(--bx-white);
            }
            .badge-primary {
              background-color: var(--bx-orange);
            }
            .badge-secondary {
              background-color: var(--bx-grey-mistery);
            }
            .badge-success {
              background-color: var(--bx-green-future);
            }
            .badge-dark {
              background-color: var(--bx-black);
            }
            .badge-danger {
              background-color: var(--bx-red-medium);
            }
            .badge-info {
              background-color: var(--bx-lblue);
              color: var(--bx-black);
            }
            .badge-warning {
              background-color: var(--bx-yellow-alert);
              color: var(--bx-black);
            }
            .badge-rounded {
              border-radius: 50rem;
            }
          
      </style>
      Badge
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
